src/has.ts

```diff
@@ -4,6 +4,58 @@ export const cache: Hash<any> = Object.create(null);
 const testFunctions: Hash<() => any> = Object.create(null);
 
 /**
+ * Conditional loading of AMD modules based on a has feature test value.
+ *
+ * @param resourceId Gives the resolved module id to load.
+ * @param require The loader require function with respect to the module that contained the plugin resource in it's dependency list.
+ * @param loaded Callback to loader that consumes result of plugin demand.
+ */
+export function load(resourceId: string, require: Function, loaded: Function): void {
+	if (resourceId) {
+		require([ resourceId ], loaded);
+	}
+	else {
+		loaded();
+	}
+}
+
+/**
+ * Resolves resourceId into a module id based on possibly-nested tenary expression that branches on has feature test value(s).
+ *
+ * @param resourceId The id of the module
+ * @param toAbsMid Resolves a relative module id into an absolute module id
+ */
+export function normalize(resourceId: string, normalize: (moduleId: string) => string): string {
```

Comment:
Seems like normalize type should be an interface somewhere that is part of a `d.ts`?